pkg/storage/compliance/tests.go

```diff
@@ -140,19 +142,13 @@ func testCatalog(t *testing.T, b storage.Backend) {
 		ver := fmt.Sprintf("v1.2.%04d", i)
 		b.Save(ctx, modname, ver, mock.Mod, bytes.NewReader(zipBts), mock.Info)
 	}
-	defer func() {
```

Comment:
Is there a reason to delete this function?

Reply:
@manugupt1 yes, because we delete everything above. Look at `clearFunc()`